portfolio/src/main/java/com/google/sps/PuzzleDao.java

```diff
@@ -0,0 +1,63 @@
+package com.google.sps;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+
+/** Puzzle DAO implementation*/
+public class PuzzleDao implements Dao<Puzzle> {
+
+    private Datastore datastore;
+    private KeyFactory keyFactory;
+
+    public PuzzleDao() {
+      datastore = DatastoreOptions.getDefaultInstance().getService();
+      keyFactory = datastore.newKeyFactory().setKind("Puzzle");
+    }
+
+    @Override
+    public void create(Puzzle puzzle) {
```

Comment:
Per my comment in the other PR have this return Puzzle:
```suggestion
    public Puzzle create(Puzzle puzzle) {
```